core/src/main/java/hudson/util/ProcessTree.java

```diff
@@ -724,6 +756,23 @@ public void kill() throws InterruptedException {
                 int pid = getPid();
                 LOGGER.fine("Killing pid="+pid);
                 UnixReflection.destroy(pid);
+                // after sending SIGTERM, wait for the process to cease to exist
+                long deadline = System.nanoTime() + softkillWaitSeconds * 1000000000;
+                do {
+                    try {
+                        FileReader r = new FileReader(getFile("status"));
+                        r.close();
+                    }
+                    catch (Exception e) {
+                        break; // process probably no longer exists (status gone)
+                    }
+
+                    try {
+                        Thread.sleep(1000);
+                    }
+                    catch (InterruptedException e) {
```

Comment:
IMO it should not be suppressed. If somebody interrupts the `kill()` command, it should be interrupted and propagated. Here the logic breaks and thread may stay active forever. It is sensitive, e.g. for shutdown of Jenkins

Reply:
Very good input, thanks!